libs/orders/src/lib/order-hooks/use-order-validation.tsx

```diff
@@ -28,33 +37,120 @@ export const useOrderValidation = ({
 }: ValidationProps) => {
   const { keypair } = useVegaWallet();
 
-  const invalidText = useMemo(() => {
+  const { message, isDisabled } = useMemo(() => {
     if (!keypair) {
-      return t('No public key selected');
+      return { message: t('No public key selected'), isDisabled: true };
     }
 
     if (keypair.tainted) {
-      return t('Selected public key has been tainted');
+      return {
+        isDisabled: true,
+        message: t('Selected public key has been tainted'),
+      };
+    }
+
+    if (
+      [
+        MarketState.Settled,
+        MarketState.Rejected,
+        MarketState.TradingTerminated,
+      ].includes(market.state)
+    ) {
+      return {
+        isDisabled: true,
+        message: t(
+          `This market is ${marketTranslations(
```

Comment:
I might have left this comment in the wrong place (sorry if I did as now it's in 2 places) but ... 

If I'm reading this correctly, then I believe it should say "This market is (state) and not accepting orders", because, for example, a market that is proposed will not have been able to take orders yet, ever.